charmcraft/commands/build.py

```diff
@@ -161,13 +143,44 @@ def build_charm(self, bases_config: BasesConfiguration) -> str:
         """
         logger.debug("Building charm in %r", str(self.buildpath))
 
-        if self.buildpath.exists():
-            shutil.rmtree(str(self.buildpath))
-        self.buildpath.mkdir()
-
-        linked_entrypoint = self.handle_generic_paths()
-        self.handle_dispatcher(linked_entrypoint)
-        self.handle_dependencies()
+        build_env = dict(LANG="C.UTF-8", LC_ALL="C.UTF-8")
+        for key in [
+            "PATH",
+            "SNAP",
+            "SNAP_ARCH",
+            "SNAP_NAME",
+            "SNAP_VERSION",
+            "http_proxy",
+            "https_proxy",
+        ]:
+            if key in os.environ:
+                build_env[key] = os.environ[key]
+
+        env_flags = [f"{key}={value}" for key, value in build_env.items()]
+
+        # invoke the charm builder
+        build_cmd = [
+            "env",
+            "-i",
+            *env_flags,
+            sys.executable,
+            "-I",
+            charm_builder.__file__,
+            "--charmdir",
+            str(self.charmdir),
+            "--builddir",
+            str(self.buildpath),
+        ]
+
+        if self.entrypoint:
+            build_cmd.extend(["--entrypoint", str(self.entrypoint)])
+
+        for req in self.requirement_paths:
+            build_cmd.extend(["-r", str(req)])
+
+        retcode = polite_exec(build_cmd)
```

Comment:
I was expecting the **plugin** to execute `charm_builder` as a script (as it runs a shell thingie). But this Python code, can't import it and use it directly?

Reply:
We could, but I implemented it as a subprocess to exercise the process invocation that will be done later from the plugin. There are a few pitfalls involved and I'd like to make sure this is working before adding lifecycle processing.